workspaces/src/cargo/mod.rs

```diff
@@ -0,0 +1,86 @@
+use anyhow::anyhow;
+use async_process::Command;
+use cargo_metadata::Message;
+use std::fmt::Debug;
+use std::fs;
+use std::path::Path;
+use std::process::Stdio;
+use tracing::debug;
+
+async fn build_cargo_project<P: AsRef<Path> + Debug>(
+    project_path: P,
+) -> anyhow::Result<Vec<Message>> {
+    let output = Command::new("cargo")
```

Comment:
What do you think about reusing the target directory of the workspace instead of having isolated spaces for each contract.

`cargo build` has `--target-dir` and `cargo_metadata::Metadata` has [`target_directory`](https://docs.rs/cargo_metadata/latest/cargo_metadata/struct.Metadata.html#structfield.target_directory)

Might be worth investigating this kind of behavior as target directories tend to take up a lot of disk space, having reusable artifacts would be a +1.

Reply:
Hmm, I am not sure I understand your point. Are you suggesting we reuse the same `workspaces` artifact for multiple contracts? But an average user wouldn't have a local copy of `workspaces` on their machine to share, would they? But using `--target-dir` to point cargo to a non-default directory might actually be worthwhile to avoid collision with user-generated compilation artifacts.

Reply:
Oh, to be clear, by "workspace" I didn't mean `workspaces-rs`, but instead `cargo` "workspace". I meant, reusing the target directory of the environment compiling the contract. So all the build artifacts resides in one place.

Following the current logic, `workspaces-rs` would cd into the contract directory and run `cargo build` in there, creating a separate `target` directory at that tree depth.

My suggestion was to research potentially merging the target directories. So contracts have their artifacts in the same `target-dir` as the project compiling them. The point of the research is see how well cargo plays with having multiple projects in the same space. My gut feeling is it would behave the same as a normal cargo workspace with members. But I'm not certain 🤞🏽.

Reply:
Ah, yeah, now I understand what you mean. Indeed, this seems like a good idea, let me give it a try.

Reply:
Works like a charm! Thanks for the suggestion